toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -375,26 +398,47 @@ func CreatePartitions(diskDevPath string, disk configuration.Disk, rootEncryptio
 }
 
 // CreateSinglePartition creates a single partition based on the partition config
-func CreateSinglePartition(diskDevPath string, partitionNumber int, partitionTableType string, partition configuration.Partition) (partDevPath string, err error) {
+func CreateSinglePartition(diskDevPath string, partitionNumber int, partitionTableType string, partition configuration.Partition, partType string) (partDevPath string, err error) {
 	const (
 		fillToEndOption  = "100%"
-		mibFmt           = "%dMiB"
+		sFmt             = "%ds"
 		timeoutInSeconds = "5"
 	)
-	start := partition.Start
-	end := partition.End
+
+	logicalSectorSize, physicalSectorSize, err := getSectorSize(diskDevPath)
+	if err != nil {
+		return
+	}
+
+	start := partition.Start * MiB / logicalSectorSize
+	end := partition.End*MiB/logicalSectorSize - 1
```

Comment:
Nit: inconsistent spacing.

```suggestion
	end := partition.End * MiB / logicalSectorSize - 1
```